src/classes/CON_Phone.cls

```diff
@@ -324,14 +324,17 @@ public with sharing class CON_Phone {
             String prefix = (String.isNotBlank(namespace)) ? namespace + '__' : '';
             List<Schema.PicklistEntry> preferredPhonePicklist = UTIL_Describe.getFieldPicklistEntries('Contact', prefix+'PreferredPhone__c');
             List<String> preferredPhoneValues = new List<String>();
+            Map<String, String> preferredLabelToApi = new Map<String, String>();  
             for (Schema.PicklistEntry eachValue: preferredPhonePicklist) {
-                preferredPhoneValues.add(eachValue.getValue());
+                PreferredLabelToApi.put(eachValue.getLabel(),eachValue.getValue());
             }
-             
-            if (!preferredPhoneValues.contains(lbl)) {
+            
+            /* If the field label is not in Preferrred Phone picklist API values then check if it is in PreferredPhoneMappings, if yes then take the value from mappings
+             else keep the label as label. If the field label is in the Preferrred Phone picklist API values then assign the picklist API value that matches the field label. */
+            if (!PreferredLabelToApi.keySet().contains(lbl)) {
                 this.prefLabel = (preferredPhoneMappings.containsKey(lbl)) ? preferredPhoneMappings.get(lbl) : lbl;
             } else {            
-                 this.prefLabel = lbl;
+                 this.prefLabel = PreferredLabelToApi.get(lbl);
```

Comment:
Also for my understanding; calling `PreferredLabelToApi.get(lbl)` will return the API value, not the label, right?

Reply:
Correct!

Reply:
Do we care that the value is getting assigned to `prefLabel` even though it's the API name, or are we ok with that?

Reply:
Hmm not sure if I understood your question correctly @ceiroa, We're assigning the value of the picklist API corresponding to picklist label that matches with the field label. Please let me know if it doesn't answer your question.

Reply:
If I understand Carlos correctly, he's just concerned that the variable name "prefLabel" is a little misleading since it might actually be the API name.

I agree it is a little confusing. The comments are good here and in CON_Email.cls but I wonder if it would be nice to explain the "why" in the comments. The comments explain the code but don't necessarily explain why it is done this way.